ranges/typedranges.nim

```diff
@@ -145,3 +145,19 @@ proc `[]=`*[T, U, V](r: MutRange[T], s: HSlice[U, V], v: Range[T]) {.inline.} =
   r[s] = toOpenArray(v)
 
 proc baseAddr*[T](r: Range[T]): ptr T {.inline.} = r.start
+
+template toRange*[T](a: Range[T]): Range[T] = a
+
+proc concat*[T](v: varargs[Range[T], toRange]): seq[T] =
+  var len = 0
+  for c in v: inc(len, c.len)
+  result = newSeq[T](len)
+  len = 0
+  for c in v:
+    copyMem(result[len].addr, c.start, sizeof(T) * c.len)
```

Comment:
`copyMem` is not appropriate in generic code. The `T` type may feature a user-defined copy operations.

Check out how `=` is defined for sequences:
https://github.com/nim-lang/Nim/blob/devel/lib/core/seqs.nim#L37

Reply:
I think the equality operator also affected by this